### PR TITLE
Switch to GitHub Actions

### PR DESCRIPTION
**Pull request by desrosj**
This PR adds an example workflow for implementing the test runner using GitHub Actions.

### CODE REVIEW

File: .travis.yml
Comment (pfefferle) [inline .travis.yml:1]: I think we should not delete the travis file. The repo is only an example for hosters and they can/should decide what to use.

**Comment by pfefferle** [comment]
@desrosj can you have a look at the latest changes?

File: .github/workflows/tests.yml
Comment (desrosj) [inline .github/workflows/tests.yml:67]: The `setup-node` action now supports caching out of the box! See WordPress/gutenberg#33190 and https://github.com/WordPress/wordpress-develop/commit/0c25dca719475585359db967f7605d247e81481a.

The Cache NodeJS modules step can be removed entirely in favor of just adding `cache: npm` to the `setup-node` step.

File: .github/workflows/tests.yml
Comment (desrosj) [inline .github/workflows/tests.yml:50]: For 3rd party actions, using version numbers and ranges is OK, but there are some tradeoffs. The [Security hardening page](https://docs.github.com/en/actions/learn-github-actions/security-hardening-for-github-actions#using-third-party-actions) in the GHA docs details these.

In Gutenberg and WordPress Core, we have adopted the pattern of using the full length commit SHA followed by an inline `# v0.8.0` comment.

**Review by desrosj — COMMENTED** [review]
Thanks for running with this, @pfefferle! I haven't looked into the reason why the workflow run is failing, but left some general feedback for the workflows themselves.

File: .github/workflows/tests.yml
Comment (pfefferle) [reply .github/workflows/tests.yml:67]: good point.

**Comment by pfefferle** [comment]
@desrosj the test fails, because it is not configured properly. This repo should only be a whitepaper/example for hosters, how they can run the unit-tests with different action or CI/CD services.

**Comment by desrosj** [comment]
@pfefferle I've made a handful of updates to polish the workflow. I think it's in a great state. I do think there is one thing we need to discuss before merging.

Since the workflow is an example for how hosts can run the tests using GHA, I think it should be disabled by default. I've added a condition in the workflow to never run, and documented that the implementer should remove that if GHA is preferred.

**Comment by pfefferle** [comment]
@desrosj I have problems with the caching of `actions/setup-node`:

```
Found in cache @ /opt/hostedtoolcache/node/14.17.6/x64
/opt/hostedtoolcache/node/14.17.6/x64/bin/npm config get cache
/home/runner/.npm
Error: Dependencies lock file is not found in /home/runner/work/***/***. Supported file patterns: package-lock.json,yarn.lock
```

**Comment by pfefferle** [comment]
> Since the workflow is an example for how hosts can run the tests using GHA, I think it should be disabled by default. I've added a condition in the workflow to never run, and documented that the implementer should remove that if GHA is preferred.

I like the idea! 👍

**Comment by desrosj** [comment]
> @desrosj I have problems with the caching of `actions/setup-node`:
> 
> ```
> Found in cache @ /opt/hostedtoolcache/node/14.17.6/x64
> /opt/hostedtoolcache/node/14.17.6/x64/bin/npm config get cache
> /home/runner/.npm
> Error: Dependencies lock file is not found in /home/runner/work/***/***. Supported file patterns: package-lock.json,yarn.lock
> ```

Ah, yes a lock file must be present for the NPM caching to configure correctly. But I'm not sure this can be configured because the `npm install` step happens in the middle of `prepare.php`. For now, I'm thinking we just remove the caching aspect.

**Comment by desrosj** [comment]
Updated! I think this one is looking pretty good!

**Comment by pfefferle** [comment]
works like a charm 👍

**Review by pfefferle — APPROVED** [review]
LGTM!

**Comment by pfefferle** [comment]
Thanks a lot @desrosj !

**Comment by getsource** [comment]
Thanks so much @pfefferle and @desrosj !! 🎉🎉🎉

**Pull request merged**
